portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -302,8 +302,7 @@ else if (PATH_PORTLET_FILE_ENTRY.equals(pathArray[0])) {
 
 					long imageCompanyId = image.getCompanyId();
 
-					if ((imageCompanyId != userCompanyId) &&
```

Comment:
I'm not opposed to removing this, though.  There's no harm in it, just possibly an extra check.